src/KnowbaseItem.php

```diff
@@ -126,6 +126,14 @@ public function canViewItem(): bool
             return true;
         }
 
+        if (
+            $this->fields["allow_access_using_token"]
+            && isset($_GET['token'])
+            && hash_equals($_GET['token'], $this->fields["token"])
```

Comment:
My bad, edited my first suggestion but it was too late.
Known string must be first according to documentation.

```suggestion
            && hash_equals($this->fields["token"], $_GET['token'])
```